x/wasm/internal/types/types.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+package wasmd.x.wasmd.v1;
+
+option go_package = "github.com/CosmWasm/wasmd/x/wasmd/internal/types";
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "types/types.proto";
+
+message MsgStoreCode{
+    option (gogoproto.goproto_getters) = false;
+
+    bytes sender = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    // WASMByteCode can be raw or gzip compressed
+    bytes wasm_byte_code = 2 [(gogoproto.customname) = "WASMByteCode", (gogoproto.moretags) = "yaml:\"wasm_byte_code\""];
+    // Source is a valid absolute HTTPS URI to the contract's source code, optional
+    string source = 3;
+    // Builder is a valid docker image name with tag, optional
+    string builder = 4;
+}
+
+message MsgInstantiateContract{
+    option (gogoproto.goproto_getters) = false;
+
+    bytes sender = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    // Admin is an optional address that can execute migrations
+    bytes admin = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 code = 3 [(gogoproto.moretags) = "yaml:\"code_id\""];
```

Comment:
why only `yaml:"code_id"` and not `json:"code_id"`?

(We can also rename the field if you wish)

Reply:
We don't use `yaml` tags with any of our other protobuf types. I don't know why this was set in the migration branch but it clearly does not set the json name as we would need it. I will replace it with `[(gogoproto.jsontag) = "code_id"]`.
Renaming the attribute would be better indeed as we also use code_id in ContractInfo but I am afraid of merge conflicts for now. It does not hurt as long as it is rendered correct in json. good 👁️